desk/app/chat.hoon

```diff
@@ -1876,21 +1887,39 @@
         ?-  -.content.memo
             %notice  ca-core
             %story
-          =/  new-message-yarn  (ca-message-hark memo p.content.memo p.p.d)
           =/  from-me  =(author.memo our.bowl)
-          =?  cor  &(want-loud-notify !from-me)
-            (emit (pass-hark new-message-yarn))
-          ?.  ?&  !=(author.memo our.bowl)
-                  |(!=(~ replying.memo) (mentioned q.p.content.memo our.bowl))
+          =/  is-mention  (mentioned q.p.content.memo our.bowl)
+          =/  to-us  |(!=(~ replying.memo) is-mention)
+          ::  if from me or don't want notifications, bail immediately
+          ?:  |(from-me &(!want-soft-notify !want-loud-notify))  ca-core
+          ?.  to-us
+            ::  if it's not a mention or reply and we want all notifications,
+            ::  send regular message notification
+            =?  cor  want-loud-notify
+              =/  =new-yarn:ha
+              %^  ca-spin  memo  ~
+              :~  [%ship author.memo]
+                  ': '
+                  (flatten q.p.content.memo)
               ==
-            ca-core
-          ?:  (mentioned q.p.content.memo our.bowl)
-            =/  new-yarn  (ca-mention-hark memo p.content.memo p.p.d)
-            =?  cor  &(want-soft-notify !want-loud-notify)
               (emit (pass-hark new-yarn))
             ca-core
+          ::  since the following are "to-us" notifications we don't have
+          ::  to check want notify anymore
+          ::
+          ::  if it's a mention send regular notification and bail
+          ?:  is-mention
+            =/  =new-yarn:ha
+            %^  ca-spin  memo  `ti
+            :~  [%ship author.memo]
+                ' mentioned you :'
+                (flatten q.p.content.memo)
```

Comment:
Not related to your changes, but maybe a good opportunity to fix this? Notifications come out like `~sampel-palnet mentioned you :sample text`, with the spacing around the semi-colon being incorrect.

```suggestion
            :~  [%ship author.memo]
                ' mentioned you: '
                (flatten q.p.content.memo)
```